src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -27,95 +37,161 @@ export class EditorComponent implements OnInit {
     this.ctx.scale(this.scaleFactor, this.scaleFactor); 
 
     //  Initalize transparent black image data to use for mask.
-    this.maskData = new ImageData(imgWidth,  imgHeight);
+    this.maskImageData = new ImageData(imgWidth,  imgHeight);
     
     this.image.onload = () => {
       this.ctx.drawImage(this.image, 0, 0, imgWidth, imgHeight);
       this.hiddenCtx.drawImage(this.image, 0, 0)
     }
   }
 
-  /** Returns the calculated scale for the image loaded. */
+  /**  Returns the calculated scale for the image loaded. */
   public getScaleFactor(): number {
     return this.scaleFactor;
   }
 
-  /** Returns the original images data for reference in mask making. */
-  public getImgData(): ImageData {
+  /**  Returns the original images data for reference in mask making. */
+  public getOriginalImageData(): ImageData {
     return this.hiddenCtx.getImageData(0,0, this.image.width, this.image.height);
   }
 
-  /** Returns black transparent ImageData for single mask image. */
-  public getMaskData(): ImageData {
-    return this.maskData;
+  /**  Returns black transparent ImageData for single mask image. */
+  public getMaskImageData(): ImageData {
+    return this.maskImageData;
   }
 
-  /** Returns the scaled images data for reference printing scaled image after mask. */
+  /**  Returns the scaled images data for reference printing scaled image after mask. */
   public getScaledData(): ImageData {
     return this.ctx.getImageData(0, 0, this.image.width * this.scaleFactor, this.image.height * this.scaleFactor);
   }
+
+  /** 
+   *  Clears canvas to get mask Image as url to store and redraws
+   *    original image for possibility user makes more edits. 
+   * @return Url for the mask image to be stored in blobstore.  
+   */
+  async getMaskBlob(): Promise<void> {
+    //  Clear canvas and put mask data to return mask as Image, 
+    this.hiddenCtx.clearRect(0,0, this.hiddenCanvas.nativeElement.width, this.hiddenCanvas.nativeElement.height);
+    this.hiddenCtx.putImageData(this.maskImageData, 0, 0);
+
+    let maskUrl = this.hiddenCanvas.nativeElement.toDataURL();
+    this.blobMask = await fetch(maskUrl).then(response => response.blob());
+    this.blobMask.lastModifiedDate = new Date();
+    this.blobMask.name = this.parentName + 'Mask.png';
+
+    //  Redraw original image if user adds more to mask
+    this.hiddenCtx.clearRect(0,0, this.hiddenCanvas.nativeElement.width, this.hiddenCanvas.nativeElement.height);
+    this.hiddenCtx.drawImage(this.image, 0, 0);
+  }
+
+  /** 
+   *  Initializes Form group and data as new
+   *  Initializes @param projectId
+   */
+  private initMaskForm() {
+    this.uploadMaskForm = new FormGroup({
+      maskName: new FormControl(),
+      labels: new FormControl(),
+    });
+
+    this.formData = new FormData();
+  }
+
+ /** 
+  *  Builds ImageBlob to be appended to form and posted.
+  *    projectIdIn: string, 
+  *    imageNameIn: string, 
+  *    modeIn: string, 
+  *    imageIn: any = '',
```

Comment:
It's best to not use the `any` type in TypeScript. Do we know what type this thing is going to be? If so, you should make it have that type. If you want to denote that nothing is stored, it would be better to make it nullable, e.g.

`imageIn: Foo | null = null,`

Reply:
Technically, when the user uploads an image from their computer the the type is a file and then when they upload the mask it is a blob type. So I didn't want to limit the type. But I tried it with type Blob and it works so I will change that.

Reply:
If you need multiple types then I think you can use a type union.